includes/platform-checkout/class-platform-checkout-utilities.php

```diff
@@ -29,4 +29,16 @@ public function should_enable_platform_checkout( $gateway ) {
 
 		return $is_platform_checkout_eligible && $is_platform_checkout_enabled && ! $disable_for_subscription;
 	}
+
+	/**
+	 * Generates a hash based on the store's blog token, merchant ID, and the time step window.
+	 *
+	 * @return string
+	 */
+	public function get_platform_checkout_request_signature() {
+		$store_blog_token = \Jetpack_Options::get_option( 'blog_token' );
+		$time_step_window = floor( time() / 30 );
+
+		return hash_hmac( 'sha512', $store_blog_token . \Jetpack_Options::get_option( 'id' ) . $time_step_window, $store_blog_token );
```

Comment:
Is there any benefit of including the key in the "plaintext" in the HMAC here? If this were an encryption algorithm, that would probably have some unintended, un-good side-effects; though, in HMAC I don't think it would necessarily break anything, since the key is not `XOR`'ed directly against the hash input in the algo. Just wondering, because I don't think I've seen it used this way before. 
```suggestion
		return hash_hmac( 'sha512', \Jetpack_Options::get_option( 'id' ) . $time_step_window, $store_blog_token );
```

Reply:
Also, is there any reason why we can't use something like [`get_current_blog_id()`](https://developer.wordpress.org/reference/functions/get_current_blog_id/) to get the current blog ID instead of querying the database for Jetpack options? `get_current_blog_id()` just uses the global variable, which should be marginally faster than making a database query.

Are the Jetpack options more reliable or more all encompassing for things like network/multi-sites, etc. ?
```suggestion
		return hash_hmac( 'sha512', get_current_blog_id() . $time_step_window, $store_blog_token );
```

Reply:
I had just added all the pieces we were using into the "data" part of the HMAC. But you're right - there's no need to include it there. I'll make the corresponding adjustment on the other end as well.